### PR TITLE
Think I got it?

### PR DESCRIPTION
**Pull request by abelmark**
Comfort: 3
Doneness: 4?

### CODE REVIEW

**Comment by abelmark** [comment]
My brain hurts.

File: google_shopping_functions.js [hunk 1]
```diff
@@ -1,30 +1,82 @@
 //define getItemsCount function
 //accepts full item data
 //returns the length of the items array
+var data = require("./products.json")
+
 var getItemsCount = function(itemData){
+	for(items in itemData){
```
Comment (jsakalys) [inline google_shopping_functions.js:7]: Here, we are just looking for a count (the length of the entire array), so we aren't actually looping through anything. You can take out the for loop and just keep the return statement here.

Comment (abelmark) [inline google_shopping_functions.js:7]: Noted.

File: google_shopping_functions.js [hunk 2]
```diff
@@ -1,30 +1,82 @@
...
+		if(items[i].product.author.name.toLowerCase().indexOf(author) != -1){
+			matchedItems.push(items[i]);
+		}
+	}
+	return matchedItems;
+}
+
+//4.  Search items by availability
+
+var getItemsByAvailability = function(items){
```
Comment (jsakalys) [inline google_shopping_functions.js:47]: To make this function more versatile, you could pass your availability status as an argument, like this:

```
var getItemsByAvailability = function(items, status) {
...
if (items[i].product.inventories[0].availability === status) {
```

**Comment by jsakalys** [comment]
Nice work!

**Pull request closed without merging**